src/signature.sh

```diff
@@ -0,0 +1,348 @@
+        shift
+        ;;
+    esac
+  done
+}
+
+function signature_help()
+{
+  if [[ "$1" == --help ]]; then
+    include "${KW_LIB_DIR}/help.sh"
+    kworkflow_man 'handle-trailer'
+    return
+  fi
+  printf '%s\n' 'kw handle-trailer:' \
+    '  handle-trailer (--add-signed-off-by | -s) (<name>) [<patchset> | <sha>] - Add Signed-off-by' \
+    '  handle-trailer (--add-reviewed-by | -r) (<name>) [<patchset> | <sha>] - Add Reviewed-by' \
+    '  handle-trailer (--add-acked-by | -a) (<name>) [<patchset> | <sha>] - Add Acked-by' \
+    '  handle-trailer (--add-fixes | -f) [<fixed-sha>] [<patchset> | <sha>] - Add Fixes' \
+    '  handle-trailer (--verbose) - Show a detailed output'
+}
+
+# This function registers a signature operation into
+# trailer_operations that will be properly excuted later.
+# It attempts to use the user's name and email configured
+# if a signature is not passed as argument and gives
+# an error if they are not properly set with git config.
+#
+# @keyword Holds a string that is a keyword for a specific operation.
+# @signature Holds a string like 'NAME <EMAIL>' defining the signature.
+function register_signature_operation()
+{
+  local keyword="$1"
+  local signature="$2"
+
+  local operation_key="ADD_${keyword}"
+  local signature_key="${keyword}_LINE"
+  local formated_output
+
+  options_values["$operation_key"]=1
+  options_values["$signature_key"]="${TRAILER_TAGS[$keyword]} "
+
+  # Use default from git config if no argument was given
+  if [[ ! "$signature" ]]; then
+    formated_output="$(format_name_email_from_user)"
+    if [[ "$?" -gt 0 ]]; then
+      return 1
+    fi
+    options_values["$signature_key"]+="$formated_output"
+  else
+    options_values["$signature_key"]+="$(str_strip "$signature")"
+  fi
+
+  trailer_operations+=("$operation_key")
+}
+
+# This function receives a commit SHA and verifies if it's
+# a valid commit reference. If it is, then it outputs the
+# appropriate formatted message. Else it returns an error code.
+#
+# @sha Holds either a commit hash or pointer
+function format_fixes_message()
+{
+  local sha="$1"
+  local formatted_message
+
+  # Check if given value is a valid commit reference
+  if [[ $(git cat-file -t "$sha" 2> /dev/null) != 'commit' ]]; then
+    return 22 # EINVAL
+  fi
+
+  # The 'Fixes:' trailer line must follow a format defined by Linux Kernel developers.
+  # Fixes: e21d2170f366 ("video: remove unnecessary platform_set_drvdata()")
+  formatted_message=$(git log -1 "$sha" --oneline --abbrev-commit --abbrev=12 \
+    --format="%h (\\\"%s\\\")")
+
+  printf '%s' "$formatted_message"
+}
+
+# It gets user.name and user.email from git's configuration.
+# If either the name or email are not configured then this
+# function will return an error code. Otherwise it will output
+# the formatted name and email.
+function format_name_email_from_user()
+{
+  local user_name
+  local user_email
+
+  user_name="$(git config user.name)"
+  user_email="$(git config user.email)"
```

Comment:
@matheustavares do you think we have a better way to get this information?

Reply:
No, I think that's the intended way for scripts too :)